packages/graphql/src/types/app.ts

```diff
@@ -0,0 +1,158 @@
+import {
+  Question,
+  QuestionDisplayMode,
+  QuestionInstance,
+  QuestionType,
+} from '@klicker-uzh/prisma'
+
+export type AvatarSettings = {
+  skinTone: string
+  eyes: string
+  mouth: string
+  hair: string
+  accessory: string
+  hairColor: string
+  clothing: string
+  clothingColor: string
+  facialHair: string
+}
+
+export type QuestionResponseChoices = {
+  choices: number[]
+}
+
+export type QuestionResponseValue = {
+  value: string
+}
+
+export type QuestionResponse = QuestionResponseChoices | QuestionResponseValue
+
+export type QuestionResultsChoices = {
+  choices: Record<string, number>
+}
+
+export type QuestionResultsOpen = {
+  [x: string]: {
+    count: number
+    value: string
+    correct?: boolean
+  }
+}
+
+export type QuestionResults = QuestionResultsChoices | QuestionResultsOpen
+
+export type Choice = {
+  ix: number
+  value: string
+  correct?: boolean
+  feedback?: string
+}
+
+export type QuestionOptionsChoices = {
+  choices: Choice[]
+}
+
+export type QuestionOptionsNumerical = {
+  unit?: string | null
+  accuracy?: number
+  placeholder?: string
+  restrictions?: {
+    min?: number
+    max?: number
+  }
+  solutionRanges?: {
+    min?: number | null
+    max?: number | null
+  }[]
+}
+
+export type QuestionOptionsFreeText = {
+  solutions?: string[]
+  restrictions?: {
+    maxLength?: number | null
+  }
+}
+
+export type QuestionOptions =
+  | QuestionOptionsChoices
+  | QuestionOptionsNumerical
+  | QuestionOptionsFreeText
+
+interface IQuestionData<
+  Type extends QuestionType,
+  Options extends QuestionOptions
```

Comment:
Why do we need to extend here and cannot directly use the QuestionType or QuestionOptions as generic types? Doesn't this work for technical reasons or do we actually want to allow more attributes than specified in those types?

Reply:
this tells TS that the generic type corresponds to QuestionType, but we use a different name for the generic as there could be naming conflicts otherwise (when we, e.g., use QuestionType in the type: Type of the interface, it could mean either the generic or the prisma type.

Reply:
that makes sense, thanks for the explanation :)